File: lib/op/interpreter/opcall/params/param/socket/validator.js
class Validator {
  /**
   * validates value against constraints
   * @param {String} value
   * @return {Array<Error>}
   */
  validate (value) {
    if (!value) {
      return [new Error('socket required')]
    }

    return []
  }
}

// export as singleton
module.exports = new Validator()

File: sdks/go/opspec/interpreter/call/op/params/param/socket/validator.go
package socket

import (
	"errors"
	"github.com/opctl/opctl/sdks/go/model"
)

type Validator interface {
	Validate(
		value *model.Value,
	) []error
}

func NewValidator() Validator {
	return _validator{}
}

type _validator struct {
}

// Validate validates a value against a string parameter
func (vdt _validator) Validate(
	value *model.Value,
) []error {
	// handle no value passed
	if nil == value || nil == value.Socket {
		return []error{errors.New("socket required")}
	}

	return []error{}
}
